engine/lib/users.php

```diff
@@ -564,8 +564,9 @@ function elgg_users_setup_entity_menu($hook, $type, $return, $params) {
 		$return = array(\ElggMenuItem::factory($options));
 	} else {
 		$return = array();
-		if (isset($entity->location)) {
-			$location = htmlspecialchars($entity->location, ENT_QUOTES, 'UTF-8', false);
+		if (isset($entity->location) && !empty($entity->location)) {
```

Comment:
Trying to write a regression test, and it seems we need to filter the array and remove any non-string and empty values. Thoughts?